### PR TITLE
[#5699] initialize bytesBuf structs before use (master)

### LINKED ISSUE #5699: IRODS 4.2.9 crashes randomly at startup
- [ ] master
- [x] 4-2-stable
---------

### iRODS Version, OS and Version

We use irods 4.2.9 in a containerized setup, with mysql as backend database, and the audit plugin enabled (to AMQP).

See https://github.com/kuleuven/irods-public for a stripped down version of our container set up, which can hopefully be used for reproducing.

### What did you try to do?

Run irods 4.2.9 in a good shape, upgrading from a perfect working 4.2.8.

### Expected behavior

It should work.

### Observed behavior (including steps to reproduce, if applicable)

There is about a 50% chance that the irods process won't start correctly, resulting in a server listening at port 1247, even responding to HEARTBEAT, but not answering any other query in a reasonable fashion.

Stacktraces such as the following occur in rodsLog: (entire logs are to be found at https://github.com/kuleuven/irods-public/tree/main/logs)

```
Caught signal [11]. Dumping stacktrace and exiting

Dumping stack trace
<0>     Offset:         Address: 0x7fad695b7423 /lib/libirods_server.so.4.2.9(+0xae1423) [0x7fad695b7423]
<1>     Offset:         Address: 0x7fad670c6630 /lib64/libpthread.so.0(+0xf630) [0x7fad670c6630]
<2>     Offset:         Address: 0x7fad697b3607 /lib/libirods_server.so.4.2.9(+0xcdd607) [0x7fad697b3607]
<3>     Offset: 0x25    Address: 0x7fad697b6105 std::__1::__function::__func<irods::error (*)(boost::any, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > >&), std::__1::allocator<irods::error (*)(boost::any, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > >&)>, irods::error (boost::any, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > >&)>::operator()(boost::any&&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > >&)
<4>     Offset: 0x55b   Address: 0x7fad697b47cb irods::re_serialization::serialize_parameter(boost::any, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > >&)
<5>     Offset: 0x6fd   Address: 0x7fad60d19cbd exec_rule(std::__1::tuple<>&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<boost::any, std::__1::allocator<boost::any> >&, irods::callback)
```

The irodsReServer cannot connect to the main process and just gives up "DEBUG: RE server is going to sleep.".

Repeatedly starting a docker container with the irods server with exact the same configuration, will sometimes work and sometimes not. See the `./run.sh` script in the above mentioned repo. The `./run.postgres.sh` script runs the same against a postgres database.

We established that:

* changing the database to postgres seems to solve the problem in most cases, but not entirely. At least once a postgres container crashed in the same way.
* disabling the audit plugin results seems to fix the problem entirely, we did not see problems at all when disabled.
* putting the irods version back to 4.2.8 (parameter in the run scripts) makes the problem go away.
* we had also occurrences where irods could start up, but e.g. "iadmin lu" would fail in 50% of the cases.

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Fixes #5699

### CODE REVIEW

File: plugins/network/ssl/libssl.cpp [hunk 1]
```diff
@@ -871,6 +871,7 @@ irods::error ssl_agent_start(
                                 // =-=-=-=-=-=-=-
                                 // call interface to read message body
                                 bytesBuf_t msg_buf;
```
Comment (trel) [inline plugins/network/ssl/libssl.cpp:873]: `bytesBuf_t msg_buf{};`

instead?

**Comment by SwooshyCueb** [comment]
@trel I think it's ready for \#, yeah?

**Comment by trel** [comment]
so with this, everything behaves as expected?    if so, yes, please # it.

**Comment by SwooshyCueb** [comment]
\#'d

**Review by trel — APPROVED** [review]

**Pull request merged**